assets/prototype/domain/shared/predicates/dateUtilities/addSub.ts

```diff
@@ -0,0 +1,53 @@
+import {
+	addDays,
+	addHours,
+	addISOWeekYears,
+	addMilliseconds,
+	addMinutes,
+	addMonths,
+	addQuarters,
+	addSeconds,
+	addWeeks,
+	addYears,
+	toDate,
+} from 'date-fns';
+
+import { toInteger } from '../../../../application/utilities/converters/number';
+
+type IntervalType =
+	| 'days'
+	| 'hours'
+	| 'ISOWeekYears'
+	| 'milliseconds'
+	| 'minutes'
+	| 'months'
+	| 'quarters'
+	| 'seconds'
+	| 'weeks'
+	| 'years';
+
+const addMapping = {
+	days: addDays,
+	hours: addHours,
+	ISOWeekYears: addISOWeekYears,
+	milliseconds: addMilliseconds,
+	minutes: addMinutes,
+	months: addMonths,
+	quarters: addQuarters,
+	seconds: addSeconds,
+	weeks: addWeeks,
+	years: addYears,
+};
+
+export const add = (interval: IntervalType, date: Date, amount: number) => {
+	const func = addMapping[interval];
+	return func(date, amount);
+};
+
+export const sub = (interval: IntervalType, dirtyDate: Date, dirtyAmount: number) => {
+	const func = addMapping[interval];
+	const amount = toInteger(dirtyAmount);
```

Comment:
This might be the answer: https://github.com/date-fns/date-fns/blob/master/src/subDays/test.js#L19

Reply:
`add` function is already doing that internally

Reply:
My intention was to avoid importing the equivalent of these:
```
        addDays,
	addHours,
	addISOWeekYears,
	addMilliseconds,
	addMinutes,
	addMonths,
	addQuarters,
	addSeconds,
	addWeeks,
	addYears,
```
***
And when I took a look what [`sub*` functions](https://github.com/date-fns/date-fns/blob/master/src/subDays/index.js#L33) are I realized that we can just swap the addition functions and get what we need.